### PR TITLE
Support for setting Viewpoint on MapView

### PR DESCRIPTION
**Pull request by gunt0001**
Adds a new parameter and type `MapViewpointOperation` for setting the viewpoint of the composable `MapView()`. This takes the form of a sealed class with subtypes that correspond to the view-based `MapView` viewpoint functions, and an extension function that executes them. Also adds a microapp which displays the different types of viewpoint operations.

### CODE REVIEW

**Review by puneet-pdx — APPROVED** [review]
@hud10837 good work. Looks good, just a comment about doc and pending name change discussion. Approving this to move the pr forward.

File: microapps/MapViewSetViewpointApp/app/src/main/java/com/arcgismaps/toolkit/mapviewsetviewpointapp/screens/MainScreen.kt [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+import com.arcgismaps.mapping.ArcGISMap
+import com.arcgismaps.mapping.BasemapStyle
+import com.arcgismaps.mapping.Bookmark
+import com.arcgismaps.mapping.Viewpoint
+import com.arcgismaps.mapping.view.AnimationCurve
+import com.arcgismaps.toolkit.geocompose.MapView
+import com.arcgismaps.toolkit.geocompose.MapViewpointOperation
+
+/**
+ * Displays a composable [com.arcgismaps.toolkit.geocompose.MapView] and permits setting the viewpoint
```
Comment (shubham7109) [inline microapps/MapViewSetViewpointApp/app/src/main/java/com/arcgismaps/toolkit/mapviewsetviewpointapp/screens/MainScreen.kt:57]: Don't think the explicit package name is needed here: 
```suggestion
 * Displays a composable [MapView] and permits setting the viewpoint
```

File: microapps/MapViewSetViewpointApp/app/src/main/java/com/arcgismaps/toolkit/mapviewsetviewpointapp/screens/MainScreen.kt [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+ * using options in a dropdown menu. The dropdown menu options represent different methods of setting
+ * the viewpoint, and all of the center on the same location. A circular progress indicator is displayed
+ * over the map while an operation is in progress.
+ */
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun MainScreen() {
+    val arcGISMap by remember { mutableStateOf(ArcGISMap(BasemapStyle.ArcGISStreets)) }
+    var mapViewpointOperation: MapViewpointOperation? by remember {
```
Comment (shubham7109) [inline microapps/MapViewSetViewpointApp/app/src/main/java/com/arcgismaps/toolkit/mapviewsetviewpointapp/screens/MainScreen.kt:66]: Maybe make this into a one-liner?

File: microapps/MapViewSetViewpointApp/app/src/main/java/com/arcgismaps/toolkit/mapviewsetviewpointapp/screens/MainScreen.kt [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+                    }
+
+                    SetViewpointDropdownMenu(
+                        onSetViewpointOperation = {
+                            mapViewpointOperation = it
+                        },
+                        expanded = actionsExpanded
+                    ) {
+                        actionsExpanded = false
+                    }
```
Comment (shubham7109) [inline microapps/MapViewSetViewpointApp/app/src/main/java/com/arcgismaps/toolkit/mapviewsetviewpointapp/screens/MainScreen.kt:96]: I would be more in favor of using the named argument here, as it shows when is the actionsExpanded being set to false:

```suggestion
                    SetViewpointDropdownMenu(
                        onSetViewpointOperation = { mapViewpointOperation = it },
                        expanded = actionsExpanded,
                        onDismissRequest = {
                            actionsExpanded = false
                        }
                    )
```

File: microapps/MapViewSetViewpointApp/app/src/main/java/com/arcgismaps/toolkit/mapviewsetviewpointapp/screens/MainScreen.kt [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+            modifier = Modifier
+                .padding(innerPadding)
+                .fillMaxSize(),
+            contentAlignment = Alignment.Center
+        ) {
+            MapView(
+                Modifier.fillMaxSize(),
+                arcGISMap = arcGISMap,
+                viewpointOperation = mapViewpointOperation
+            )
```
Comment (shubham7109) [inline microapps/MapViewSetViewpointApp/app/src/main/java/com/arcgismaps/toolkit/mapviewsetviewpointapp/screens/MainScreen.kt:111]:
```suggestion
            MapView(
                modifier = Modifier.fillMaxSize(),
                arcGISMap = arcGISMap,
                viewpointOperation = mapViewpointOperation
            )
```

File: microapps/MapViewSetViewpointApp/app/src/main/java/com/arcgismaps/toolkit/mapviewsetviewpointapp/screens/MainScreen.kt [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+            if (showProgressIndicator) {
+                CircularProgressIndicator()
+            }
+        }
+    }
+}
+
+@Composable
+fun SetViewpointDropdownMenu(
```
Comment (shubham7109) [inline microapps/MapViewSetViewpointApp/app/src/main/java/com/arcgismaps/toolkit/mapviewsetviewpointapp/screens/MainScreen.kt:120]: Add doc for this compose function

File: microapps/MapViewSetViewpointApp/app/src/main/java/com/arcgismaps/toolkit/mapviewsetviewpointapp/screens/MainScreen.kt [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+    onSetViewpointOperation: (MapViewpointOperation) -> Unit,
+    modifier: Modifier = Modifier,
+    expanded: Boolean,
+    onDismissRequest: () -> Unit
+) {
```
Comment (shubham7109) [inline microapps/MapViewSetViewpointApp/app/src/main/java/com/arcgismaps/toolkit/mapviewsetviewpointapp/screens/MainScreen.kt:125]: I think the API guidelines mention using the modifier as the first parameter of a compose function, and lambdas grouped at the bottom. 
```suggestion
    modifier: Modifier = Modifier,
    expanded: Boolean,
    onSetViewpointOperation: (MapViewpointOperation) -> Unit,
    onDismissRequest: () -> Unit
) {
```

File: microapps/MapViewSetViewpointApp/app/src/main/java/com/arcgismaps/toolkit/mapviewsetviewpointapp/screens/MainScreen.kt [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+    val items = remember {
```
Comment (shubham7109) [inline microapps/MapViewSetViewpointApp/app/src/main/java/com/arcgismaps/toolkit/mapviewsetviewpointapp/screens/MainScreen.kt:126]: A less generic name would be nice here
```suggestion
    val viewpointOperationsList = remember {
```

File: microapps/MapViewSetViewpointApp/app/src/main/java/com/arcgismaps/toolkit/mapviewsetviewpointapp/screens/MainScreen.kt [hunk 8]
```diff
@@ -0,0 +1,176 @@
...
+            Viewpoint(sofia, scale)
+        )
+    }
+    val sofiaBounds = remember {
+        GeometryEngine.bufferOrNull(sofia, 0.083) ?: error("Couldn't buffer point")
+    }
+    DropdownMenu(
+        expanded = expanded,
+        onDismissRequest = onDismissRequest,
+        modifier = modifier
```
Comment (shubham7109) [inline microapps/MapViewSetViewpointApp/app/src/main/java/com/arcgismaps/toolkit/mapviewsetviewpointapp/screens/MainScreen.kt:145]: Move modifier as the first parameter

File: microapps/MapViewSetViewpointApp/app/src/main/java/com/arcgismaps/toolkit/mapviewsetviewpointapp/screens/MainScreen.kt [hunk 9]
```diff
@@ -0,0 +1,176 @@
...
+    ) {
+        items.forEach {
+            val name = it
```
Comment (shubham7109) [inline microapps/MapViewSetViewpointApp/app/src/main/java/com/arcgismaps/toolkit/mapviewsetviewpointapp/screens/MainScreen.kt:148]:
```suggestion
            val viewpointOperationName = it
```

**Review by shubham7109 — COMMENTED** [review]
The PR mostly looks good to me. Few comments...

File: microapps/MapViewSetViewpointApp/app/src/main/java/com/arcgismaps/toolkit/mapviewsetviewpointapp/screens/MainScreen.kt (hunk 6 above)
Comment (hud10837) [reply microapps/MapViewSetViewpointApp/app/src/main/java/com/arcgismaps/toolkit/mapviewsetviewpointapp/screens/MainScreen.kt:125]: I believe it's meant to be the first optional parameter, so I have changed it to be that.

**Comment by hud10837** [comment]
Thanks @shubham7109 I believe I have addressed all of your comments

**Review by gunt0001 — COMMENTED** [review]
@hud10837 - looks great, a few comments, mostly doc related.

**Review by shubham7109 — APPROVED** [review]
Looks good 👍

File: toolkit/geo-compose/src/main/java/com/arcgismaps/toolkit/geocompose/MapViewpointOperation.kt [hunk 10]
```diff
@@ -0,0 +1,208 @@
...
+     * navigation.
+     *
+     * @property viewpoint the new viewpoint
+     * @property durationSeconds the duration of the animation in seconds
+     * @property curve the animation curve to apply
+     * @since 200.3.0
+     */
+    public class Animate(
+        public val viewpoint: Viewpoint,
+        public val durationSeconds: Float = 0f,
```
[the remark this answers is not in the document]
Comment (hud10837) [reply toolkit/geo-compose/src/main/java/com/arcgismaps/toolkit/geocompose/MapViewpointOperation.kt]: I was able to confirm that the Core value is 0.25f, so I have changed it to that.

**Review by gunt0001 — COMMENTED** [review]
@hud10837 - looks good 👍

**Pull request merged**